src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -1108,6 +1107,15 @@ export class TerminalService implements ITerminalService {
 		return undefined;
 	}
 
+	private _optionsAreBasic(options?: ICreateTerminalOptions): boolean {
+		if (!options) {
+			return true;
+		} else if (Object.entries(options).length > 2) {
+			return false;
+		}
+		return Array.from(Object.keys(options)).filter(k => k !== 'target' && k !== 'isSplitTerminal').length === 0;
+	}
+
```

Comment:
Can we just check if `options?.config` is falsy?

Reply:
Not sure what you mean by that, but we want to allow the options on line 1116 or this will not work in many cases that we want it to

Reply:
We just need to make sure we're considering the default profile here and then checking if it's a contributed profile, so I think we can do that simpler than `_optionsAreBasic` by checking is `ICreateTerminalOptions.config` is undefined

Reply:
try it - you'll see why that does not work.

Reply:
we should ignore the forceSplit and target options. if any others are present, we do not want to use the default profile.